av.h

```diff
@@ -108,6 +108,26 @@ Perl equivalent: C<my @array;>.
 
 #define newAV()	MUTABLE_AV(newSV_type(SVt_PVAV))
 
+/*
+=for apidoc newAV_alloc_x
+
+Similar to newAV(), but the SV* array is also allocated.
+
+This is similar to but more efficient than doing:
+
+    AV *av = newAV();
+    av_extend(av, key);
+
+Note that these the actual size requested is allocated. This is unlike
+av_extend(), which enforces a minimum size of 3, and always does a +1
+to the requested/minimum size when allocating.
```

Comment:
It adds 1 to the _key_, but that's not the "requested/minimum size" but the index of the maximum element. So the minimum key is 3, but minimum size is 4 (covering indices 0-3).

Reply:
Yes, good point. I'll try to improve the wording.